File: substrings.js
class Solution{
    findSubstring(s, words) {
        /*
        :type s: str
        :type words: List[str]
        :rtype: List[int]
        */

        var combinations = [];
        var combination = '';
        var reverse_combination = '';
        var results = [];

        words.forEach(function(word) {
          combination += word;
        });

        words.reverse();
        words.forEach(function(word) {
          reverse_combination += word;
        });

        combinations.push(combination);
        combinations.push(reverse_combination);

        combinations.forEach(function(word) {
          var max_index = (s.length - 1);
          var start = 0;
          var end = word.length;
          while ((end <= max_index)) {
              if ((word === s.slice(start, end))) {
                  results.push(start);
              }
              start += 1;
              end += 1;
          }
        });

        return results;
    }
}
function main() {
    var s, solution, words;
    solution = new Solution();
    s = "barfoothefoobarman";
    words = ["foo", "bar"];
    console.log(solution.findSubstring(s, words));
    s = "wordgoodstudentgoodword";
    words = ["word", "student"];
    console.log(solution.findSubstring(s, words));
}

main();

File: substrings.py
class Solution(object):
    def findSubstring(self, s, words):
        """
        :type s: str
        :type words: List[str]
        :rtype: List[int]
        """
        combinations = []
        combination = str()
        reverse_combination = str()

        results = []

        for word in words:
            combination += word

        for word in words[::-1]:
            reverse_combination += word

        combinations.append(combination)
        combinations.append(reverse_combination)

        for word in combinations:
            max_index = len(s)-1
            start = 0
            end = len(word)

            while end <= max_index:
                if word == s[start:end]:
                    results.append(start)

                start += 1
                end += 1

        results.sort()

        return results

def main():
    solution = Solution()
    s = "barfoothefoobarman"
    words = ["foo", "bar"]
    print(solution.findSubstring(s,words))
    s = "wordgoodstudentgoodword"
    words = ["word", "student"]
    print(solution.findSubstring(s,words))

if __name__ == '__main__':
    main()
